airflow-core/src/airflow/api_fastapi/auth/managers/base_auth_manager.py

```diff
@@ -427,36 +433,243 @@ def get_authorized_dag_ids(
+        :param user: the user
+        :param method: the method to filter on
+        :param session: the session
+        """
+        stmt = select(Variable.key, Team.name).join(Team, Variable.team_id == Team.id, isouter=True)
+        rows = session.execute(stmt).all()
+        variables_by_team: dict[str | None, set[str]] = defaultdict(set)
+        for var_key, team_name in rows:
+            variables_by_team[team_name].add(var_key)
+
+        var_keys: set[str] = set()
+        for team_name, team_var_keys in variables_by_team.items():
+            var_keys.update(
+                self.filter_authorized_variables(
+                    variable_keys=team_var_keys, user=user, method=method, team_name=team_name
+                )
+            )
+
+        return var_keys
+
+    def filter_authorized_variables(
+        self,
+        *,
+        variable_keys: set[str],
+        user: T,
+        method: ResourceMethod = "GET",
+        team_name: str | None = None,
+    ) -> set[str]:
+        """
+        Filter variables the user has access to.
+
+        By default, check individually if the user has permissions to access the variable.
+        Can lead to some poor performance. It is recommended to override this method in the auth manager
+        implementation to provide a more efficient implementation.
+
+        :param variable_keys: the set of variable keys
+        :param user: the user
+        :param method: the method to filter on
+        :param team_name: the name of the team associated to the connections if Airflow environment runs in
+            multi-team mode
+        """
+
+        def _is_authorized_variable(var_key: str):
+            return self.is_authorized_variable(
+                method=method, details=VariableDetails(key=var_key, team_name=team_name), user=user
+            )
+
+        return {var_key for var_key in variable_keys if _is_authorized_variable(var_key)}
+
+    @provide_session
+    def get_authorized_pools(
+        self,
+        *,
+        user: T,
+        method: ResourceMethod = "GET",
+        session: Session = NEW_SESSION,
+    ) -> set[str]:
+        """
+        Get pools the user has access to.
+
+        :param user: the user
+        :param method: the method to filter on
+        :param session: the session
+        """
+        stmt = select(Pool.pool, Team.name).join(Team, Pool.team_id == Team.id, isouter=True)
+        rows = session.execute(stmt).all()
+        pools_by_team: dict[str | None, set[str]] = defaultdict(set)
+        for pool_name, team_name in rows:
+            pools_by_team[team_name].add(pool_name)
+
+        pool_names: set[str] = set()
+        for team_name, team_pool_names in pools_by_team.items():
+            pool_names.update(
+                self.filter_authorized_pools(
+                    pool_names=team_pool_names, user=user, method=method, team_name=team_name
+                )
+            )
+
+        return pool_names
```

Comment:
If I understand correctly, it seems the `filter_authorized_<entites>` can be consolidated with `get_authorized_<entines>` as:

```suggestion        
        for team_name, team_pool_names in pools_by_team.items():
            # extend the set with new team
            pool_names.update(team_pool_names)

            # filter out invalid pools
            for pool_name in team_pool_names:
                if not self.is_authorized_pool(
                    method=method,
                    details=PoolDetails(name=pool_name, team_name=team_name),
                    user=user
                ):
                    pools_name.remove(pool_name

        return pool_names
```

Reply:
I do not want to consolidate them. They need to be 2 different methods because this is an interface/base class. I expect sub classes to override the `filter_authorized_` methods but not the `get_authorized_` methods. One is retrieving all the entities from the DB, the other is filtering these entities based on user permissions.

Reply:
Thanks for clarifying! If that's the case, we can leave it as is.